imgix/urlbuilder.py

```diff
@@ -93,6 +95,19 @@ def create_url(self, path, params={}, opts={}):
         else:
             domain = self._domains[0]
 
+        regex = re.compile(
+                r'^(?:[a-z\d\-_]{1,62}\.){0,125}'
+                r'(?:[a-z\d](?:\-(?=\-*[a-z\d])|[a-z]|\d){0,62}\.)'
+                r'[a-z\d]{1,63}$'
+        )
+
+        if re.match(regex, domain) is None:
+            warnings.warn(
```

Comment:
What are your thoughts about warning here vs. erroring?

Reply:
I think we should actually be throwing an error. I was a bit worried of users constantly getting hit with an error without knowing how to remediate it, but that's something that should be better outlined in the documentation.